modules/ROOT/pages/migrate-client-apps.adoc

```diff
@@ -0,0 +1,149 @@
+// +
+// This property is displayed on the application page.
+
+[[migrate_with_external]]
+== Migrate Client Applications for APIs with External Client Providers
+
+When creating an application for an organization with registered external client providers, you can directly associate the application with the API that you want to consume. To reference an API, you must first retrieve the API instance ID, which is available from the API Manager UI.
+
+You can include the client ID and client secret when referencing an API. However, this only works if you using a PingFederate external client provider or to the Anypoint Platform client provider
+
+When creating applications referencing APIs associated with a provider, such as Okta or Keycloak that are registered using OpenID Dynamic registration, you cannot define client ID and client secret credentials. This is because the OpenID Connect Dynamic Client Registration specification does not include client ID and client secret when creating a client.
+
+To migrate your client application:
+
+. Open a text editor and paste the following commands:
++
+[source,text,linenums]
+----
+curl --location --request POST 'https://anypoint.mulesoft.com/exchange/api/v1/organizations/:orgId/applications?apiInstanceId=:myApiInstanceId' \
+--header 'Authorization: Bearer <omitted>' \
+--header 'Content-Type: application/json' \
+--data-raw '{
+    "clientId": "myIdpClientID",
+    "clientSecret": "myIdpClientSecret",
+    "grantTypes": [
+        "client_credentials"
+    ],
+    "name": "My Application Name",
+    "redirectUri": [
+        "https://anypoint.mulesoft.com/apiplatform/:orgId/authentication/oauth2.html",
+        "https://anypoint.mulesoft.com/apiplatform/:orgId/admin/authentication/oauth2.html",
+        "https://api-notebook.anypoint.mulesoft.com/authenticate/oauth.html"
+    ]
+}'
+----
+. In the command snippet, replace the `orgId` parameter with your organization Id, `apiInstanceId` with the API instance ID, and the content of the body with the appropriate application <<properties,properties>>. 
+. Run this command at the Command prompt.
+
+[[properties]]
+== Properties Reference for Migrating Applications
+
+In the properties of the migrate command, provide values based on the following recommendations:
+
+* `clientId`: <Your client ID>
+* `clientSecret`: <Your client secret>
+* (optional)`description`: <Informative description for the applications>
++
+This property is displayed on the application page. 
+* `name`: <Name of the application to be created>
++
+This property is displayed when selecting an application before requesting access.
+* `redirectUri`: <Array containing the redirect URIs for your application to use as an OAuth 2.0 client>
+* (optional) `url`: <URL of the client application>
++
+This property is displayed on the application page.
+* `grantTypes`: <Empty array for Anypoint Platform client provider or an array with possible values for external client providers: 
```

Comment:
This is where the format you have chosen breaks down, because it uses TWO colons (and is missing >)
```suggestion
* `grantTypes`: <Empty array for Anypoint Platform client provider or an array with possible values for external client providers>: 
```

Reply:
I suggest changing the format of this reference going forward, making all entries match this sample:
```suggestion
* `grantTypes`
+
<Empty array for Anypoint Platform client provider or an array with possible values for external client providers: 
** `Resource Owner Grant`: `password`
OR
** `Resource Owner Grant`
+
`password`
```

Reply:
Changed it to a dash.